src/lib/io/sol-gpio-common.c

```diff
@@ -74,3 +77,39 @@ sol_gpio_open(uint32_t pin, const struct sol_gpio_config *config)
 
     return gpio;
 }
+
+SOL_API enum sol_gpio_edge
+sol_gpio_edge_from_str(const char *edge)
+{
+    int len = strlen(edge);
+    enum sol_gpio_edge gpio_edge;
+    static const struct sol_str_table table[] = {
+        SOL_STR_TABLE_ITEM("none", SOL_GPIO_EDGE_NONE),
+        SOL_STR_TABLE_ITEM("rising", SOL_GPIO_EDGE_RISING),
+        SOL_STR_TABLE_ITEM("falling", SOL_GPIO_EDGE_FALLING),
+        SOL_STR_TABLE_ITEM("any", SOL_GPIO_EDGE_BOTH),
+        { }
+    };
+
+    if (!sol_str_table_lookup(table, SOL_STR_SLICE_STR(edge, len), &gpio_edge))
```

Comment:
Very minor, but you can use 'sol_str_slice_from_str()' function instead of the macro and manual strlen()